File: 0020_Valid_Parentheses/MQQM2.py
'''
  题目：
  给定一个只包括 '('，')'，'{'，'}'，'['，']' 的字符串，判断字符串是否有效。
  
  说明：
  有效字符串需满足：左括号必须用相同类型的右括号闭合。左括号必须以正确的顺序闭合。
  注意空字符串可被认为是有效字符串。
  
  参考：
  https://blog.csdn.net/qq_17550379/article/details/80723003
'''
class Solution:
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stk = list()
        m = {'(':')','[':']','{':'}'}
        for i in s:
            if i == '(' or i == '[' or i == '{':
                stk.append(i)
            else:
                if len(stk) == 0:
                    return False
                top = stk.pop()
                if m[top] != i:
                    return False
        if len(stk) == 0:
            return True
        
        return False   

File: problems/0070_Climbing_Stairs/MQQM2.cpp
/*
  题目：
  假设你正在爬楼梯。需要 n 阶你才能到达楼顶。
  每次你可以爬 1 或 2 个台阶。你有多少种不同的方法可以爬到楼顶呢？
  
  参考：
  https://blog.csdn.net/haimianxiaodao/article/details/52608536?utm_source=blogxgwz6
*/
class Solution {
public:
    int climbStairs(int n) {          
        int n_1 = 1;
        int n_2 = 0;
        
        int tmp;
        for(int i=1; i<=n; i++){       
            tmp = n_1+n_2;
            n_2 = n_1;
            n_1 = tmp;    
        }
    
        return tmp;
    }
};
